test/kinetics/kineticsFromYaml.cpp

```diff
@@ -10,7 +10,26 @@
 
 using namespace Cantera;
 
-TEST(Reaction, ElementaryFromYaml)
+TEST(ReactionRate, ModifyArrheniusRate)
+{
+    auto sol = newSolution("gri30.yaml");
+    AnyMap rxn = AnyMap::fromYamlString(
+        "{equation: N + NO <=> N2 + O,"
+        " rate-constant: [-2.70000E+13 cm^3/mol/s, 0, 355 cal/mol],"
+        " negative-A: true}");
+
+    auto R = newReaction(rxn, *(sol->kinetics()));
+    auto ER = dynamic_cast<ElementaryReaction3&>(*R);
+
+    auto rr0 = std::dynamic_pointer_cast<ArrheniusRate>(ER.rate());
+    EXPECT_TRUE(rr0->allow_negative_pre_exponential_factor);
+    rr0->allow_negative_pre_exponential_factor = false;
+
+    auto rr1 = std::dynamic_pointer_cast<ArrheniusRate>(ER.rate());
+    EXPECT_FALSE(rr1->allow_negative_pre_exponential_factor);
```

Comment:
This latter test seems redundant to me. It's pretty clear that `rr0` and `rr1` must be pointers to the same `ArrheniusRate` object, isn't it?